lib/sparql_to_sw_solr/instance_solr_doc.rb

```diff
@@ -25,14 +33,62 @@ def solr_doc_hash
         return if CKEY_BLACKLIST.include?(@ckey)
         doc = {}
         doc[:id] = instance_uri_to_ckey
-        doc
+        build_doc_titles(doc)
       end
     end
 
     private
 
+    def build_doc_titles(doc)
+      doc[:id] = instance_uri_to_ckey
+      doc[:title_245a_search] = primary_triples(['mainTitle'], title_query_str)
+      doc[:title_245_search] = primary_triples(['rdf-schema#label'], title_query_str)
+      doc[:title_245a_display] = primary_triples(['mainTitle'], title_query_str)
+      doc[:title_display] = primary_triples(%w(mainTitle subtitle), title_query_str)
+      doc[:title_full_display] = "#{primary_triples(%w(mainTitle subtitle), title_query_str)} / " \
+                                 "#{primary_triples(['responsibilityStatement'], responsibility_query_str)}"
+      doc
+    end
+
     def instance_uri_to_ckey
       @ckey ||= self.class.instance_uri_to_ckey(@instance_uri)
     end
+
+    def sparql
+      @sparql ||= SPARQL::Client.new(SPARQL_URL)
+    end
+
+    def title_query_str
+      "#{BF_NS_DECL}
+        SELECT ?p ?o WHERE {
+            <#{instance_uri}> bf:title ?t .
+            ?t ?p ?o .
+            filter not exists { ?t a bf:VariantTitle }
+        }
+      "
+      # This will be for when we get the bibframe ontology loaded into the graph:
+      # filter not exists { ?t a bf:VariantTitle ;
+      #   rdfs:subClassOf bf:VariantTitle } .
+    end
+
+    def responsibility_query_str
```

Comment:
ditto above:  it's really a constant, and it's more along the lines of `RESPONSIBILITY_SPARQL` ... and here, it should probably be `?s` for subject, not `?t` for title